controllers/managedocs_controller.go

```diff
@@ -911,6 +916,26 @@ func (r *ManagedOCSReconciler) reconcilePrometheus() error {
 	return nil
 }
 
+// reconcilePrometheusService function wait for prometheus Service
+// to start and sets appropriate annotation for 'service-ca' controller
+func (r *ManagedOCSReconciler) reconcilePrometheusService() error {
+	r.Log.Info("Reconciling Prometheus Service")
+	prometheusService := &corev1.Service{}
+	err := r.Client.Get(r.ctx,
+		types.NamespacedName{Name: prometheusServiceName, Namespace: r.namespace},
+		prometheusService)
+	// as this Service is not created explicitly by us,
+	// we don't raise any error, will wait till it is available
+	if err != nil {
+		return nil
+	}
+	// check whether the Service has appropriate annotation
+	if annotVal := prometheusService.Annotations[serviceServingCertAnnotationKeyBeta]; annotVal != prometheusServiceName {
+		prometheusService.Annotations[serviceServingCertAnnotationKeyBeta] = prometheusServiceName
+	}
```

Comment:
Some issues:
1. The annotation map might be nil
2. What you wrote here is equivalent to just setting the value
3. You are not setting the annotations via the interface

```suggestion
	anno := prometheusService.GetAnnotations()
	if (anno == nil) {
		anno = new map[string]string{}
		prometheusService.SetAnnotations()
	}
	anno[serviceServingCertAnnotationKeyBeta] = prometheusServiceName
```

The above works because a map is a ref type

Reply:
This is changed. Now we directly assign into the map (without any `nil` checking), something like this
```go
r.prometheusService.Annotations = map[string]string{
	servingCertSecretBetaAnnotationKey:  prometheusServiceName,
	servingCertSecretAlphaAnnotationKey: prometheusServiceName,
}
```
The thought behind this was we are reconciling to bring the resource to the expected state and the above is the expected state.

Reply:
This is a bad practice as you are overriding whatever other annotation any other actor/user will add to the service.
Please use the suggestion from the last comment